### PR TITLE
[New Rule] Kernel Driver Load by non-root User

### PR DESCRIPTION
**Pull request by Aegrah**
## Summary
Detects the loading of a Linux kernel module by a non-root user through system calls. Threat actors may leverage Linux kernel modules to load a rootkit on a system providing them with complete control and the ability to hide from security products. As other rules monitor for the addition of Linux kernel modules through system utilities or .ko files, this rule covers the gap that evasive rootkits leverage by monitoring for kernel module additions on the lowest level through auditd_manager.

## Detection
This behavior is possible when a binary is assigned the `cap_sys_module` capability.
```
driver where host.os.type == "linux" and event.dataset == "auditd_manager.auditd" and 
event.action == "loaded-kernel-module" and auditd.data.syscall in ("init_module", "finit_module") and user.id != "0"
```

### CODE REVIEW

File: rules/linux/persistence_kernel_driver_load_by_non_root.toml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+min_stack_comments = "New fields added: required_fields, related_integrations, setup"
+min_stack_version = "8.3.0"
+updated_date = "2024/01/10"
+
+[rule]
+author = ["Elastic"]
+description = """
+Detects the loading of a Linux kernel module by a non-root user through system calls. Threat actors may leverage Linux
+kernel modules to load a rootkit on a system providing them with complete control and the ability to hide from security
+products. As other rules monitor for the addition of Linux kernel modules through system utilities or .ko files, this
```
Comment (shashank-elastic) [inline rules/linux/persistence_kernel_driver_load_by_non_root.toml:14]: Curious if you would like to call out related rules in the note section something like
note = "### Related rules
 <Rule -1>
"
Just gives an overall picture to the user to enable all rules with kernel module monitoring.

**Review by shashank-elastic — APPROVED** [review]
LGTM,
Nit: Good to have related rules called out as the description contains "other rules monitor".

**Review by brokensound77 — COMMENTED** [review]
nice, the logic LGTM, but can you add resources and an investigation guide

**Review by imays11 — APPROVED** [review]

File: rules/linux/persistence_kernel_driver_load_by_non_root.toml (hunk 1 above)
Comment (Aegrah) [reply rules/linux/persistence_kernel_driver_load_by_non_root.toml:14]: This will be done in the future investigation guide!

**Pull request merged**